src/app/components/trip-details/trip-details.component.html

```diff
@@ -15,25 +45,58 @@ <h1 class="text-2xl font-bold">{{ trip.title }}</h1>
-                {{ date.getDate() }}
-              </li>
-              @if (trip.itinerary && trip.itinerary.length > 0) {
-                @for (activity of trip.itinerary; track activity) {
-                  @if (
-                    activity.startDate &&
-                    getDateWithoutTime(
-                      activity.startDate.toDate()
-                    ).valueOf() === getDateWithoutTime(date).valueOf()
-                  ) {
-                    <li>
-                      <nz-badge
-                        [nzStatus]="'warning'"
-                        class="w-11 overflow-hidden text-ellipsis text-nowrap md:w-28">
-                        {{ activity.title }}
-                      </nz-badge>
-                    </li>
-                  }
+              @if (
+                getDateWithoutTime(date).valueOf() ===
+                  getDateWithoutTime(trip.startDate).valueOf() &&
+                getDateWithoutTime(date).valueOf() ===
+                  getDateWithoutTime(trip.endDate).valueOf()
+              ) {
+                @if (date | hasActivity: trip.itinerary ?? []) {
+                  <li
+                    class="tripStartAndEndDate rounded-full bg-green-1000 text-black">
+                    {{ date.getDate() }}
+                  </li>
+                } @else {
+                  <li class="tripStartAndEndDate">
+                    {{ date.getDate() }}
+                  </li>
+                }
+              } @else if (
+                getDateWithoutTime(date).valueOf() ===
+                getDateWithoutTime(trip.startDate).valueOf()
+              ) {
+                @if (date | hasActivity: trip.itinerary ?? []) {
+                  <li
+                    class="tripStartDate rounded-full bg-green-1000 text-black">
+                    {{ date.getDate() }}
+                  </li>
+                } @else {
+                  <li class="tripStartDate">
+                    {{ date.getDate() }}
+                  </li>
+                }
+              } @else if (
+                getDateWithoutTime(date).valueOf() ===
+                getDateWithoutTime(trip.endDate).valueOf()
+              ) {
+                @if (date | hasActivity: trip.itinerary ?? []) {
+                  <li class="tripEndDate rounded-full bg-green-1000 text-black">
+                    {{ date.getDate() }}
+                  </li>
+                } @else {
+                  <li class="tripEndDate">
+                    {{ date.getDate() }}
+                  </li>
+                }
+              } @else {
+                @if (date | hasActivity: trip.itinerary ?? []) {
+                  <li class="tripDate rounded-full bg-green-1000 text-black">
+                    {{ date.getDate() }}
+                  </li>
+                } @else {
+                  <li class="tripDate">
+                    {{ date.getDate() }}
+                  </li>
```

Comment:
OR:
```
<li [class]="'tripDate' + (date | hasActivity: trip.itinerary ?? []) ? 'rounded-full bg-green-1000 text-black' : ''">
  {{ date.getDate() }}
</li>
```

Reply:
This is not an instruction but an alternative

Reply:
Another idea is:
`[class.rounded-full]="date | hasActivity: trip.itinerary ?? []"`
`[class.bg-green-1000]="date | hasActivity: trip.itinerary ?? []"`
etc